### DISCUSSION TITLE
How to make a rootable original  ISO installer or a backup of the Master system

### DISCUSSION THREAD

**Discussion by MrAsana** [MrAsana/AMBER_B1_ROS2#3] [Q&A]
## How to restore the Initial Master system, make a bootable  AMBER Master system USB installer, or a backup of your revised Master system  on your own.

Step 1,  Download the Ventoy, that is an open source tool to create bootable USB drive for ISO files.

>  Download Link:  https://www.ventoy.net/en/download.html

>  Make bootable USB drive:  https://www.ventoy.net/en/doc_start.html

Step 2, Download AMBER B1 initial control  system ISO file  from the link below.

For Ubuntu 20.04+ROS2 Foxy ONLY,  https://drive.google.com/drive/folders/1pOU09vUy5v_DGHXibgdyFncFs8cxHgq4

For Ubuntu 22.04+ROS2 Humble Version ONLY,  https://drive.google.com/file/d/18H61_vc1MCSTrzMq_Nl--WaLd_mbUxC5

Step 3, Copy ISO file to the USB Drive made by Step 1. 

Step 4,  Install this firmware to AMBER B1 Master box by this bootable USB drive.  See this step by step instruction below.
https://docs.google.com/document/d/1MU08JQNKIl4DXH71kOS0MwCSJQCvQ1__gY2SRo5ZNiA

Step 5, Update the shell file, b1.sh, master.sh, stop_master.sh by https://github.com/MrAsana/AMBER_B1_ROS2/tree/main/Scripts

### For the issue of no partition around 500M during installation,  please check the link below to fix it
https://docs.google.com/document/d/1aeMmQKioEe3Ji4QClzYlpGu5YMxJ7QPqT-3Z4ZH5u80/

**Comment by nambogit** [comment]
Unfortunately I’ve already been messing around with it a bit since I’m not too familiar with Linux and was missing certain files (eg ros2_start had to be built). I had thought we had an ISO uploaded already so I didn’t make a master backup ahead of time. Can you upload a copy of the ISO for our copy of the firmware so all the backers with our version can cross-reference? Thanks!

**Reply by MrAsana** [reply]
It's been updated, please download the right version for the Master Control Computer you have.

**Comment by MrAsana** [comment]
A userful link how to  make a iso installer to backup of your revised Master system. 

https://copyfuture.com/blogs-details/202201011456305864

**Comment by nambogit** [comment]
I’m checking in to see if you had already released the ISO somewhere - my version doesn’t have a ros2_start or a start file in my ros2_ws folder so I was hoping to get it switched to the corrected version. Thanks!

**Reply by lsascha** [reply]
had the same file missing.
But i found the content that goes into that file here: https://github.com/AmberInside/Amber_AI_ROS2/issues/12#issuecomment-1064721868
after i created the file and made it executable, i got it working with no issues so far

**Reply by nambogit** [reply]
Were you from the first batch or second batch? I tried that method and it
didn’t create the file for me, so I assumed it didn’t work for the second
batch. I’ll try again though!

**Reply by lsascha** [reply]
i don't know. i think i am from the first batch.

I have the one with a gray plastic Hub and black PC.

I think the second batch is with a wooden textured box that has the hub and PC integrated together?
Not sure though.

**Reply by MrAsana** [reply]
> I’m checking in to see if you had already released the ISO somewhere - my version doesn’t have a ros2_start or a start file in my ros2_ws folder so I was hoping to get it switched to the corrected version. Thanks!

About the script name, some of then is ros2_start.sh , some is start.sh;  It doesn't matter the name of the scripts.   
To run the script name in the directory of /home/amber/ros2_ws.

**Comment by nambogit** [comment]
I’m also from the separate hub/pc batch and it’s definitely promising that
it worked for you - thanks for the info!

**Comment by Gotjisan** [comment]
Hallo. the file AMBER_ROS2.iso is missing and can not be download.

**Reply by MrAsana** [reply]
This former ISO file is only for the Master Control Computer (Ubuntu 20.04+Foxy)when you get it before May, 2022.   We  change it to be invisible to avoid misleading subsequent users
Please leave a message to hello@amberobotics.com if you need this former version.

**Reply by Gotjisan** [reply]
Amber B1 robotic arm and gripper. Is there a datasheet to read? I need data on the weight of the gripper, the work load, and the work area arm of the of the robot.

**Comment by JoeYoMama** [comment]
Hi sorry but what is the password for the humble version ? it doesnt apear to be amber.

**Reply by majdros** [reply]
with 'a' works on my B1 master-box
